loopdb/migration_04_updates.go

```diff
@@ -0,0 +1,112 @@
+package loopdb
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/coreos/bbolt"
+)
+
+// migrateUpdates migrates the swap updates to add an additional level of
+// nesting, allowing for optional keys to be added.
+func migrateUpdates(tx *bbolt.Tx, chainParams *chaincfg.Params) error {
+	for _, key := range [][]byte{loopInBucketKey, loopOutBucketKey} {
+		rootBucket := tx.Bucket(key)
+		if rootBucket == nil {
+			return fmt.Errorf("bucket %v does not exist", key)
+		}
+
+		err := migrateSwapTypeUpdates(rootBucket)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// migrateSwapTypeUpdates migrates updates for swaps in the specified bucket.
+func migrateSwapTypeUpdates(rootBucket *bbolt.Bucket) error {
+	var swaps [][]byte
+	err := rootBucket.ForEach(func(swapHash, v []byte) error {
+		// Only go into things that we know are sub-bucket
+		// keys.
+		if v == nil {
+			swaps = append(swaps, swapHash)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Do not modify inside the for each.
```

Comment:
Nit: this comment belongs above the for each imo.

Reply:
moved